src/main/java/sg/ncl/MainController.java

```diff
@@ -456,36 +449,67 @@ public String validateDetails(
             }
 
             if (errorsFound) {
+                logger.warn("Signup new team error {}", signUpMergedForm.toString());
                 return "/signup2";
             } else {
 
-                // FIXME need to check if team exists?
                 teamFields.put("name", signUpMergedForm.getTeamName());
                 teamFields.put("description", signUpMergedForm.getTeamDescription());
                 teamFields.put("website", signUpMergedForm.getTeamWebsite());
                 teamFields.put("organisationType", signUpMergedForm.getTeamOrganizationType());
                 teamFields.put("visibility", signUpMergedForm.getIsPublic());
                 mainObject.put("isJoinTeam", false);
-                registerUserToDeter(mainObject);
+
+                try {
+                    registerUserToDeter(mainObject);
+                } catch (TeamNotFoundException e) {
+                    redirectAttributes.addFlashAttribute("message", e.getMessage());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "message" 17 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)